File: Backend/RaiseYourVoice.Domain/Entities/Post.cs
using System;
using System.Collections.Generic;
using RaiseYourVoice.Domain.Common;
using RaiseYourVoice.Domain.Enums;

namespace RaiseYourVoice.Domain.Entities
{
    public class Post : BaseEntity
    {
        public required string Title { get; set; }
        public required string Content { get; set; }
        public List<string> MediaUrls { get; set; } = new List<string>();
        public PostType PostType { get; set; }
        public required string AuthorId { get; set; }
        public int LikeCount { get; set; }
        public int CommentCount { get; set; }
        public List<string> Tags { get; set; } = new List<string>();
        public Location? Location { get; set; }
        public DateTime? EventDate { get; set; }
        public PostStatus Status { get; set; }
    }

    public class Location
    {
        public string? Address { get; set; }
        public required string City { get; set; }
        public required string Country { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
    }
}

File: Mobile/Android/app/src/main/java/al/raiseyourvoice/android/domain/model/Post.kt
package al.raiseyourvoice.android.domain.model

import java.util.Date

/**
 * Domain model representing a post in the application.
 */
data class Post(
    val id: String,
    val title: String,
    val content: String,
    val mediaUrls: List<String> = emptyList(),
    val postType: PostType,
    val authorId: String,
    val authorName: String,
    val authorProfilePicUrl: String? = null,
    val createdAt: Date,
    val updatedAt: Date? = null,
    val likeCount: Int = 0,
    val commentCount: Int = 0,
    val isLikedByCurrentUser: Boolean = false,
    val tags: List<String> = emptyList(),
    val location: Location? = null,
    val eventDate: Date? = null,
    val status: PostStatus = PostStatus.PUBLISHED
)

enum class PostType {
    ACTIVISM,
    OPPORTUNITY,
    SUCCESS_STORY
}

enum class PostStatus {
    PUBLISHED,
    DRAFT,
    REMOVED
}

data class Location(
    val latitude: Double,
    val longitude: Double,
    val address: String?,
    val city: String?,
    val country: String?
)